base-java/src/main/java/com/ibm/mq/samples/java/BasicGet.java

```diff
@@ -0,0 +1,133 @@
+package com.ibm.mq.samples.java;
+
+import com.ibm.mq.*;
+import com.ibm.mq.constants.MQConstants;
+import org.json.*;
+
+import java.io.FileReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Hashtable;
+
+public class BasicGet {
+
+    private static class MQDetails {
+        String QMGR;
+        String QUEUE_NAME;
+        String HOST;
+        String PORT;
+        String CHANNEL;
+        String USER;
+        String PASSWORD;
+        String KEY_REPOSITORY;
+        String CIPHER;
+    }
+
+    private static JSONArray endpoints;
+
+    public static void main(String[] args) {
+        loadEnv("env.json");
+
+        for (int i = 0; i < endpoints.length(); i++) {
+            System.out.println("Processing endpoint " + i);
+            JSONObject point = endpoints.getJSONObject(i);
+            MQDetails details = buildMQDetails(point);
+            processEndpoint(details);
+        }
+
+        System.out.println("Sample MQ GET application ending");
+    }
+
+    private static void loadEnv(String path) {
```

Comment:
repeated code - should be in a private class method (ie not static) in the shared base class.